command.go

```diff
@@ -189,7 +189,7 @@ func (c Command) startApp(ctx *Context) error {
 
 	// set the name and usage
 	app.Name = fmt.Sprintf("%s %s", ctx.App.Name, c.Name)
-	if c.HelpName == "" {
+	if c.HelpName != "" {
```

Comment:
Actually, I take that back, that logic appears to be intentional -- was it manifesting in an issue for you?